.circleci/template.yml

```diff
@@ -0,0 +1,978 @@
+            if [ -s _build/mim2/rel/mongooseim/log/crash.log ]; then cat _build/mim2/rel/mongooseim/log/crash.log; fi
+            tail -100 _build/mim1/rel/mongooseim/log/mongooseim.log.1
+            tail -100 _build/mim2/rel/mongooseim/log/mongooseim.log.1
+      - upload_results_to_aws
+      - publish_github_comment
+      - publish_github_summary_comment
+
+  # #########################
+  # jobs in docker containers
+  docker_image:
+    parallelism: 1
+    executor: << parameters.executor >>
+    parameters:
+      executor:
+        type: executor
+    steps:
+      - checkout
+      - prepare_for_cache
+      - setup_remote_docker
+      - restore_prod_build
+      - run:
+          name: Execute Docker image build and upload
+          command: tools/circle-build-and-push-docker.sh
+
+  docs_build_deploy:
+    parallelism: 1
+    docker:
+      - image: cimg/python:3.9.0
+    working_directory: ~/app
+    steps:
+      - checkout
+      - run:
+          name: Test that docs build
+          command: |
+            pip3 install mkdocs
+            pip3 install mkdocs-material
+            mkdocs build --strict
+      - when:
+          condition:
+            or:
+              - equal: [ master, << pipeline.git.branch >> ]
+              - << pipeline.git.tag >>
+          steps:
+            - add_ssh_keys
+            - run:
+                name: Configure Git
+                command: |
+                  git config --global user.email "mongoose-im@erlang-solutions.com"
+                  git config --global user.name "mongooseim"
+            - run:
+                name: Generate image tag
+                command: |
+                  set -x
+                  if [ -n "$CIRCLE_TAG" ]; then
+                      DOCS_TAG="$CIRCLE_TAG"
+                  elif [ "$CIRCLE_BRANCH" == "master" ]; then
+                      DOCS_TAG="latest"
+                  else
+                      DOCS_TAG=`echo $CIRCLE_BRANCH | sed 's,/,_,g'`
+                  fi
+                  env | sort
+                  echo "export DOCS_TAG=$DOCS_TAG" >> $BASH_ENV
+            - run:
+                name: Build latest
+                command: |
+                  git fetch git@github.com:esl/MongooseDocs.git gh-pages:gh-pages
+                  pip3 install mike
+                  mike deploy $DOCS_TAG --remote git@github.com:esl/MongooseDocs.git --branch gh-pages --push --rebase
+
+  build_in_docker:
+    executor: << parameters.executor >>
+    parameters:
+      executor:
+        type: executor
+      build_prod:
+        type: boolean
+        description: When enabled, prod release will be generated besides the dev ones
+        default: false
+    parallelism: 1
+    environment:
+      SKIP_RELEASE: 1
+      SKIP_COV: 0
+      SKIP_CERT_BUILD: 1
+    steps:
+      - checkout
+      - prepare_for_cache
+      - fetch_build_packages
+      - maybe_build_deps_and_cache
+      - restore_certs
```

Comment:
Maybe I'd put this one earlier, so that everything stops early if this one really goes wrong.